x-pack/plugins/triggers_actions_ui/public/application/sections/alerts_list/components/alerts_list.test.tsx

```diff
@@ -178,9 +177,25 @@ describe('alerts_list component with items', () => {
       },
       monitoring: {
         execution: {
-          history: [{ success: true }, { success: true }, { success: false }],
+          history: [
+            {
+              success: true,
+              duration: 100,
+            },
+            {
+              success: true,
+              duration: 200,
+            },
+            {
+              success: false,
```

Comment:
nice!

Reply:
for the sake of it, add test with duration who is null and not number and it does not exist in the object. just to make sure we do not crash.

Reply:
oups!!! maybe not here but where we do the calculation ;)

Reply:
It didn't seem very feasible to test that case using integration tests, so I moved the logic for the percentile calculation into it's own file and unit tested that instead. This actually works well since @chrisronline suggested we should move some of that monitoring logic into it's own file.